keystone/identity/backends/ldap.py

```diff
@@ -106,8 +113,13 @@ def update_user(self, user_id, user):
             # values are already equal.
             user['enabled'] = not user['enabled']
             old_obj['enabled'] = not old_obj['enabled']
+        try:
+            self.user.update(user_id, user, old_obj)
+        except ldap.CONSTRAINT_VIOLATION as e:
```

Comment:
The code itself is in the driver for the ldap backend, if there is no ldap backend this code is never called.